lib/Zef/Distribution.pm6

```diff
@@ -66,31 +66,32 @@ class Zef::Distribution does Distribution is Zef::Distribution::DependencySpecif
         return %normalized;
     }
 
-    # make matching dependency names against a dist easier
-    # when sorting the install order from the meta hash
-    method depends-specs       {
-        my $depends = system-collapse($.depends);
-        my $deps := $depends ~~ Hash
-            ?? $depends<runtime build>.grep(*.defined).grep(*.<requires>).map(*.<requires>).map(*.Slip)
-            !! $depends;
-
-        $deps.grep(*.defined).map({ Zef::Distribution::DependencySpecification.new($_) }).grep(*.name);
+    # 'new-depends' refers to the hash form of `depends`
+    has $!new-depends-cache;
+    method !new-depends($type) {
+        return Empty unless $.depends ~~ Hash;
```

Comment:
Could use a multi to separate Hash from non-Hash

Reply:
It’s a private method